src/EditorFeatures/Test2/Rename/InlineRenameTests.vb

```diff
@@ -56,6 +56,30 @@ Namespace Microsoft.CodeAnalysis.Editor.UnitTests.Rename
             End Using
         End Function
 
+        <WpfFact>
+        <Trait(Traits.Feature, Traits.Features.Rename)>
+        Public Async Function RenameLambdaDiscard() As Task
+            Using workspace = CreateWorkspaceWithWaiter(
+                    <Workspace>
+                        <Project Language="C#" CommonReferences="true">
+                            <Document><![CDATA[
+class C
+{
+    void M()
+    {
+        C _ = null;
+        System.Func<int, string, int> f = (int _, string [|$$_|]) => { _ = null; return 1; };
```

Comment:
@jasonmalinowski @dpoeschl I'm not sure if this is the correct way to test this, but when I test manually, I see that Rename isn't offered on discards, which is what I'd expect (`GetDeclaredSymbol`  return `null` for discards, so renaming wouldn't be possible as it's not even clear what symbol would be renamed).

Reply:
I think blocking is a completely reasonable decision: I didn't want us marching on with a symbol that had no name or a strange name and then crashing. That said if it's blocking we need to put a test at a different layer, but that's not a problem. We'll clean this one up.